dataprofiler/data_readers/csv_data.py

```diff
@@ -104,23 +104,163 @@ def _check_and_return_options(self, options):
+            sniffer.preferred = preferred
+            try:
+                dialect = sniffer.sniff(data_as_str)
+                quotechar = dialect.quotechar
+            except csv.Error as exc:
+                quotechar = '"'
+
+        # Evaluate vocab, reviewing rows and columns
+        delimiter = None
+        validated_proposed_delimiters = {}
+        for proposed_delim in ordered_vocab[0:12]:
+            proposed_dataset = []
+            for row in data_as_str.split('\n'):
+                proposed_cells = list(csv.reader(
+                    [row], delimiter=proposed_delim, quotechar=quotechar))[0]
+                
+                proposed_row = []
+                for i in range(len(proposed_cells)):
+                    proposed_cell = proposed_cells[i]
+                    size_of_cell = len(proposed_cell)
+                    cell_type = data_utils.detect_cell_type(proposed_cell)
+            
+                    proposed_row.append({
+                        "cell": proposed_cells[i],
+                        "type": cell_type,
+                        "size": size_of_cell
+                    })
+                proposed_dataset.append(proposed_row)
+
+            col_types = {}            
+            prior_col_count = None
+            incorrect_delimiter_flag = False
+            cell_type_safe_flag = True
+            row_end_safe_flag = False
+            row_start_safe_flag = False
+
+            proposed_delim_type = data_utils.detect_cell_type(proposed_delim)
+        
+            # Reverse to start at bottom
+            # Fewer columns are okay, as long as earlier in file
+            proposed_dataset.reverse()
+            for proposed_row in proposed_dataset:
+
+                # Ensure there's a prior_col_count
+                if not prior_col_count:
+                    prior_col_count = len(proposed_row)
+
+                # Ensure's it's not a description row, then ensures cells consistent
+                if len(proposed_row) != prior_col_count and len(proposed_row) > 1:
+                    incorrect_delimiter_flag = True
+                    break
+
+
+                # Evalutes each column for potential issues
+                # e.g. '3' being a seperator, while a single row of integers
+                cell_content = []
+                cell_types = []
+                prior_cell_type = None
+                for col_id in range(len(proposed_row)):
+                    proposed_cell = proposed_row[col_id]
+
+                    col_types[col_id] = proposed_cell['type']
+
+                    # Handle if alpha character are seperator                     
+                    # NOTE: delimiter needs two ajoining cells to flag
+                    if proposed_cell['type'] in ['str', 'none'] \
+                       and prior_cell_type in ['str', 'none']:
+                        if proposed_delim.isalpha():
+                            cell_type_safe_flag = False
```

Comment:
Can we break from the loop if any of these flags would cause failure?

Reply:
Doesn't meaningfully impact results

Reply:
Added some breaks